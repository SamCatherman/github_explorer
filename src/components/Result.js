import React from "react";

const Result = props => (
  <div className="result-container">
    <h4>{props.username}</h4>
    <p>{props.fullName}, {props.location}</p>
    <p>{props.type}</p>
    <a href={props.page}>Visit {props.username}</a>
    <img src={props.img} alt={props.username}/>
  </div>
)

export default Result;
