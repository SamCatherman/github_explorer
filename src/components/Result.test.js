import React from "react";
import Result from "./Result";
import { shallow } from "enzyme";

describe("<Result />", () => {
  it("renders correctly", () => {
    let wrapper = shallow(<Result />);
    expect(wrapper).toMatchSnapshot();
  });
});
