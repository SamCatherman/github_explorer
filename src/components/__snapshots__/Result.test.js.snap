// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Result /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Result />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4 />,
        null,
        <img
          alt={undefined}
          src={undefined}
/>,
        <p />,
        <p>
          Public Repositories: 
</p>,
        <a
          href={undefined}
>
          Visit 
</a>,
        <a
          href="mailto:undefined"
>
          Contact 
</a>,
      ],
      "className": "result-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h4",
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": undefined,
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Public Repositories: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Public Repositories: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Visit ",
            undefined,
          ],
          "href": undefined,
        },
        "ref": null,
        "rendered": Array [
          "Visit ",
          undefined,
        ],
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Contact ",
            undefined,
          ],
          "href": "mailto:undefined",
        },
        "ref": null,
        "rendered": Array [
          "Contact ",
          undefined,
        ],
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4 />,
          null,
          <img
            alt={undefined}
            src={undefined}
/>,
          <p />,
          <p>
            Public Repositories: 
</p>,
          <a
            href={undefined}
>
            Visit 
</a>,
          <a
            href="mailto:undefined"
>
            Contact 
</a>,
        ],
        "className": "result-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h4",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": undefined,
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Public Repositories: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Public Repositories: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Visit ",
              undefined,
            ],
            "href": undefined,
          },
          "ref": null,
          "rendered": Array [
            "Visit ",
            undefined,
          ],
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Contact ",
              undefined,
            ],
            "href": "mailto:undefined",
          },
          "ref": null,
          "rendered": Array [
            "Contact ",
            undefined,
          ],
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
